PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/actions/SparkSubmitJobAction.kt

```diff
@@ -109,6 +119,14 @@ open class SparkSubmitJobAction : AzureAnAction() {
             model.isClusterSelectionEnabled = true
         }
     }
+
+    override fun getServiceName(event: AnActionEvent): String? {
+        return SelectSparkApplicationTypeAction.getSelectedSparkApplicationType().value
+    }
+
+    override fun getOperationName(event: AnActionEvent?): String {
```

Comment:
Unused parameter `event`?

Reply:
This is an override method. We don't need `event` in this implementation here but there are some override implementation that needs this parameter. BTW, maybe there are some other designs that can avoid this situation, do you have any suggestions?

---
In reply to: [297082820](https://github.com/microsoft/azure-tools-for-java/pull/3201#discussion_r297082820) [](ancestors = 297082820)

Reply:
I am still confused that: getServiceName and getOperationName need a event argument, I find all implementations  in this PR don't use it(am I missing anything?). Do all the calls you keep in one AnActionEvent are in the same thread? If so, you can leverage a Contenxt  class set in a thread local map.

Reply:
Hi Andy, Thanks for review the code! Please find the definition of `SparkJobRunAction::getServiceName`. We need to get project from event and then determine service name based on spark application type of runcongifurationSettings.
```
    override fun getServiceName(event: AnActionEvent): String {
        val project = event?.project ?: return super.getServiceName(event)
        val runManagerEx = RunManagerEx.getInstanceEx(project)
        val selectedConfigSettings = runManagerEx.selectedConfiguration
        return (selectedConfigSettings?.configuration as LivySparkBatchJobRunConfiguration).sparkApplicationType.value
    }
```

---
In reply to: [298440076](https://github.com/microsoft/azure-tools-for-java/pull/3201#discussion_r298440076) [](ancestors = 298440076)